### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by drunkymunky** [HASwitchPlate/openHASP#277] [Q&A]
Hi, i have Espduino-32 aka. Wemos “TTGo” D1 R32 development board with a waveshare 4" tft touch shield, (https://haswitchplate.github.io/openHASP-docs/0.6.2/displays/waveshare-tft-touch-shield/#d1-r32) there is a precompile build for this?
thank you

**Comment by fvanroie** [comment]
I have added the `d1-r32-espduino32` artifacts to the development [*nightly builds*](https://github.com/HASwitchPlate/openHASP/actions) as they were not available yet.
From now on you can find the latest pre-build development firmware in the `d1-r32-espduino32.zip` file.

Please test it. Make sure the 3 dip switches are in the ON position. If all is well, I will also include it in the next release.

**Reply by drunkymunky** [reply]
thanks for the answer, sorry but I'm a beginner in this, it's my first project with esp32 + tft, I wanted to try to integrate with home assistant that I have been using for several years, the file I have to flash is d1-r32-waveshare_ili9486_full_4MB_v0.7.0.bin ? I will come home next week and take a test, thanks again

**Comment by drunkymunky** [comment]
Hi, i flash with the last nightly build but the device not discoverde from hassio, the log report is :

2022-01-12 10:14:25 ERROR (MainThread) [custom_components.openhasp.config_flow] Version mismatch! Your plate: 0.7.0 - openHASP Component: 0.6.x
2022-01-12 10:14:55 ERROR (MainThread) [custom_components.openhasp.config_flow] Discovery Only

i use 0.6.4 custom component version install manualy, follow step by step how-to guide, am I doing something wrong?
thank you

**Reply by fvanroie** [reply]
Could please try to comment [this line](https://github.com/HASwitchPlate/openHASP-custom-component/issues/87#issuecomment-1015941574) in `config_flow.py` of the CC?

```python
            raise data_entry_flow.AbortFlow("mismatch_version")
```

I found that it fixes dicoverability of `0.7` plates. Please post your findings in the custom-component issue [#87](https://github.com/HASwitchPlate/openHASP-custom-component/issues/87).

**Reply by drunkymunky** [reply]
last seventh I tried to edit the const.py file, I changed Line 5

Version
MAJOR = "0"
MINOR = "7"

before it was

Version
MAJOR = "0"
MINOR = "6"

i also edited 
manifest.json line 8
"version": "0.7.0",
before it was
"version": "0.6.3",
used the last version of CC, https://github.com/HASwitchPlate/openHASP-custom-component/releases/tag/0.6.4,

I don't know if what I did is right but it worked, and integration with hassio is perfect, even alpha 0.7 works great with the D1 R32 development board with a waveshare 4 "tft touch shield, now run with this version https://github.com/HASwitchPlate/openHASP/actions/runs/1710209796

later i will upload what i managed to do with this amazing openhasp, thanks @fvanroie 

sorry for my bad english, write from italy :)
